core/processor/batch_market_instructions_processor.go

```diff
@@ -124,10 +131,28 @@ func (p *BMIProcessor) ProcessBatch(
 	// these need to be determinitistic
 	idgen := idgeneration.New(determinitisticID)
 
+	failedMarkets := map[string]error{}
+	for _, umm := range batch.UpdateMarginMode {
+		err := p.validator.CheckUpdateMarginMode(umm)
+		if err == nil {
+			var marginFactor num.Decimal
+			if umm.MarginFactor == nil || len(*umm.MarginFactor) == 0 {
+				marginFactor = num.DecimalZero()
+			} else {
+				marginFactor = num.MustDecimalFromString(*umm.MarginFactor)
```

Comment:
MustDecimal here will panic with a malformed string isn't it?
```suggestion
				marginFactor = num.MustDecimalFromString(*umm.MarginFactor)
```